Paco-Server/src/com/google/sampling/experiential/server/ACLHelper.java

```diff
@@ -0,0 +1,92 @@
+package com.google.sampling.experiential.server;
+
+import java.util.List;
+
+import com.google.common.collect.Lists;
+import com.pacoapp.paco.shared.util.SearchUtil;
+
+import net.sf.jsqlparser.statement.Statement;
+
+public class ACLHelper {
+  public static String getModifiedQueryBasedOnACL(String selectSql, String loggedInUser,
+                                                  List<Long> adminExperimentsinDB) throws Exception {
+    Statement selStat = SearchUtil.getJsqlStatement(selectSql);
+    return getModifiedQueryBasedOnACL( selStat, loggedInUser,adminExperimentsinDB);
+  }
+  public static String getModifiedQueryBasedOnACL(Statement selectSql, String loggedInUser,
+                                                  List<Long> adminExperimentsinDB) throws Exception {
+    String adminExpIdCSV = adminExperimentsinDB.toString();
+    String loggedInUserWithQuotes = "'"+ loggedInUser +"'";
+    boolean ownData = false;
+    boolean ownExpt = false;
+    adminExpIdCSV = adminExpIdCSV.replace("[", "");
+    adminExpIdCSV = adminExpIdCSV.replace("]", "");
+
+    List<String> userSpecifiedWhoValues = SearchUtil.retrieveUserSpecifiedConditions(selectSql, "who");
+    List<String> userSpecifiedExpId = SearchUtil.retrieveUserSpecifiedConditions(selectSql, "experiment_id");
+    List<Long> userSpecifiedExpIdValues = convertToLong(userSpecifiedExpId);
+
+    // if user querying own data
+    for (String s : userSpecifiedWhoValues) {
+      if (s.equalsIgnoreCase(loggedInUserWithQuotes)) {
+        ownData = true;
+      } else {
+        ownData = false;
+      }
+    }
+
+    // if user querying own experiments
+    for (Long s : userSpecifiedExpIdValues) {
```

Comment:
same problem as who example above. The ownExpt should be false if any are not in the administered list.